mbuild/formats/hoomd_snapshot.py

```diff
@@ -12,7 +12,57 @@
 hoomd = import_("hoomd")
 hoomd.data = import_("hoomd.data")
 
-__all__ = ["to_hoomdsnapshot"]
+__all__ = ["to_hoomdsnapshot", "from_snapshot"]
+
+
+def from_snapshot(snapshot, scale=1.0):
+    """Convert a hoomd.data.Snapshot or a gsd.hoomd.Snapshot to an
+    mbuild Compound.
+
+    Parameters
+    ----------
+    snapshot : hoomd.data.SnapshotParticleData or gsd.hoomd.Snapshot
+        Snapshot from which to build the mbuild Compound.
+    scale : float, optional, default 1.0
+        Value by which to scale the length values
+
+    Returns
+    -------
+    comp : mb.Compound
+    """
+    comp = mb.Compound()
+    bond_array = snapshot.bonds.group
+    n_atoms = snapshot.particles.N
+
+    # There will be a better way to do this once box overhaul merged
+    # Only orthogonal boxes allowed
+    try:
+        # gsd
+        box = snapshot.configuration.box
+        comp.box = mb.Box(lengths=box[:3] * scale)
+    except AttributeError:
+        # hoomd
+        box = snapshot.box
+        comp.box = mb.Box(lengths=np.array([box.Lx,box.Ly,box.Lz]) * scale)
```

Comment:
An `AttributeError` here doesn't necessarily imply that this is a hoomd snapshot object. I recommend using specific checks when needed and let the more generic code path raise an exception if it is going to:
```suggestion
    if 'SnapshotSystemData_float' in dir(hoomd._hoomd) and isinstance(snapshot, hoomd._hoomd.SnapshotSystemData_float):
        # hoomd v2
        box = snapshot.box
        comp.box = mb.Box(lengths=np.array([box.Lx,box.Ly,box.Lz]) * scale)
    else:
        # gsd / hoomd v3
        box = snapshot.configuration.box
        comp.box = mb.Box(lengths=box[:3] * scale)
```